feat/deena-emma/journal popup

faquanly opened the pull request:
close #6 
# Documentation For Journal Popup (updated 5/22)

## Role of Popup

The role of this component is to have a journal popup feature so that the user can input a journal entry in their dev notebook. 

<img width="853" alt="image" src="https://github.com/cse110-sp24-group7/cse110-sp24-group7/assets/142956234/48fd57ee-fde8-4dd2-a3bb-48be03a1cb81">

## Intended Purpose

The journal popup is intended to open whenever the user clicks add new journal. A popup will display and the user has the ability to name their journal, give it a date, and add in their actual journal entry. Once they finish they can click the "done" button to save, or the "x" button to cancel. 



Note: 
- on submit button there is an entry_id, entry_title, entry_content, and creation_date saved to localStorage (and outputted on console)
- (EDITED) to follow same formatting and styling as the task popup

We implemented this through a seperate HTML, JS, and CSS file. 


## Requirements

The popup should allow the user to input information for each journal entry including date, title, and content. The background should blur to improve readability and clarity. It should be simple and straightforward to use from a UI perspective.


## Testing

For testing our journal popup component, we used the LiveServer to periodically check any changes that were made and the effects of those changes.

#### Authors

- Deena Pederson

- Emma Nguyen

deenap23 commented on source/project/components/app.js:
```diff
@@ -0,0 +1,99 @@
+   */
+  connectedCallback() {
+    // event listener to form submission
+    setTimeout(() => {
```
this code was being called before the html was put into the DOM by the constructor so I added a timeout. Is there a better way to do this?

SamvritSrinath replied:
Maybe have another function that puts the html into the DOM, not sure whether this was resolved?

deenap23 commented on source/project/components/app.js:
```diff
@@ -0,0 +1,99 @@
+        .addEventListener("submit", this.onSubmit.bind(this));
+    }, 3000);
+
+    // const closeButton = document.getElementById("closeBtn"); old
```
In progress, working on getting the close button functionality (same issue that hashim and Matteo had)

mialyssa reviewed (approved):
Functionality wise, from Monday's meeting everything looks great and works as expected! The last documentation update is from 5/18, so I am not sure if it reflects the most recent stylistic changes; namely the done button alignment. Otherwise, It looks good to merge!

n2singh reviewed (approved):
Everything looks good, the styling is up to par with the other formats, unless there is something I am missing I think we are good to merge!

SamvritSrinath commented:
Just an FYI —> vscode files should be ignored and not pushed to the main branch(I see that you are using it for testing, the GitHub tests will not run on the cloud). 

functionally things seem to be fine but we shouldn’t be fetching components from the frontend(I see that you are using the fetch API rather than creating a shadow component, could you embed this in the app.js and rename that to journal-popup.js and embed both this and the stylesheet within a folder?

SamvritSrinath reviewed (changes requested):
Refactor Code structure according to latest comment —> components should be placed in their own folders with their own stylesheets. VScode files should be removed.

faquanly commented:
We changed `app.js` to `journal-popup.js` and added the unit test `journal-popup.test.js` in `./__test__ /` directory
<img width="1170" alt="Screenshot 2024-05-23 at 7 12 57 PM" src="https://github.com/cse110-sp24-group7/cse110-sp24-group7/assets/78623809/d0d89098-cb52-488c-9668-b3ab545059cf">

arnavkamdar reviewed (commented):
I think it looks pretty good apart from the vs code file like Samvrit mentioned. I'd also add that the test HTML you use should probably be named a little more formally than testingDeena.html, but that's a pretty minor issue to fix.
